### DISCUSSION TITLE
gzip static assets when uploading to S3

### DISCUSSION THREAD

**Discussion by andybee** [architect/architect#1253] [Feature Requests]
We configure API Gateway to gzip response payloads to reduce the size on transfer, but if we also gzip'ed assets on their way to the static S3 bucket we could reduce the amount of data over the wire between S3 and Lambda.

Low priority as it's unlikely to yield huge wins, but could be a "nice to have" improvement.

**Comment by filmaj** [comment]
I _think_ the solution is to set the `Content-Encoding` header on objects we upload to S3 (along with also compressing the content itself). If this is correct, I wonder if there would be any side effects here... are there any situations where we would _not_ want to set this header on upload? Presumably it would set in stone the `Content-Encoding` header on responses to all requests for the object stored in S3 as well.

**Comment by ryanblock** [comment]
I am aware of potential for side effects with compression + edge caching behind CloudFront distributions, something I was working on recently. Was going to check back on this issue as I dug into this eventually!

**Comment by ryanblock** [comment]
Minor update: Architect Functions + HTTP Proxy now support this as of https://github.com/architect/functions/pull/283

**Comment by ryanblock** [comment]
Update: as of Arc 10.3, all static assets are now brotli compressed, as well as non-binary responses delivered via `@architect/functions`!
